src/ScatterUser.ts

```diff
@@ -112,4 +106,27 @@ export class ScatterUser extends User {
         e)
     }
   }
+
+  private authenticate(challenge: string, resolve): void {
+    this.scatter.authenticate(challenge).then(async (signature) => {
+      const publicKey = this.getPublicKey(challenge, signature)
+      const myKeys = await this.getKeys()
+      let resolvedValue = false
+      for (const key of myKeys) {
+        if (key === publicKey) {
+          resolvedValue = true
+        }
+      }
+      resolve(resolvedValue)
+    })
+  }
+
+  private getPublicKey(challenge: string, signature: string): string {
+    const ellipticSignature = Signature.fromString(signature).toElliptic()
+    const ellipticHashedStringAsBuffer = Buffer.from(ellipticEc.hash().update(challenge).digest(), 'hex')
+    const ellipticRecoveredPublicKey =
+      ellipticEc.recoverPubKey(ellipticHashedStringAsBuffer, ellipticSignature, ellipticSignature.recoveryParam, 'hex')
+    const ellipticPublicKey = ellipticEc.keyFromPublic(ellipticRecoveredPublicKey)
+    return PublicKey.fromElliptic(ellipticPublicKey).toString()
```

Comment:
This file is a great place to start forming and testing theories about whether or not we can keep the API as it was and not have clients have to just use the elliptic library rather than sticking to our eosjs interface.